### PR TITLE
Google signin

### PR DESCRIPTION
**Pull request by DimaKelek**
1. Google sign in
2. Redux for authorization

### CODE REVIEW

File: agoraStreaming/src/Screens/Auth/helpers/googleSignIn.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import {GoogleSignin} from '@react-native-google-signin/google-signin';
+import {Dispatch, SetStateAction} from 'react';
+
+export const onGoogleButtonPress = async () => {
+  try {
+    const {idToken} = await GoogleSignin.signIn();
+    const googleCredential = auth.GoogleAuthProvider.credential(idToken);
+    await auth().signInWithCredential(googleCredential);
+  } catch (error) {
+    console.error(error);
```
Comment (koverchik) [inline agoraStreaming/src/Screens/Auth/helpers/googleSignIn.ts:11]: I think we can display error on the screen.

File: agoraStreaming/src/Navigation/Tab/index.tsx [hunk 2]
```diff
@@ -41,13 +45,31 @@ export const BottomTabs = () => {
...
+    if (userInfo) {
+      const {uid, email, displayName} = userInfo;
+      dispatch(setUser({displayName, uid, email}));
+    }
+  };
+
+  useEffect(() => {
+    GoogleSignin.configure({
+      webClientId:
+        '1088468777432-7titji4f8tsu0oorpqfibu469sl8jvnj.apps.googleusercontent.com',
```
Comment (evgenyshenets91) [inline agoraStreaming/src/Navigation/Tab/index.tsx:64]: move that string to separate const

File: agoraStreaming/src/Redux/actions/AuthActions/types.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import {AuthDataType} from '../../reducers/Auth/types';
+
+export type AuthActionsType = ReturnType<SetUserType>;
+
+export enum AuthActions {
+  SET_USER = 'SET-USER',
```
Comment (evgenyshenets91) [inline agoraStreaming/src/Redux/actions/AuthActions/types.ts:6]: please use value the same as the key - SET_USER

File: agoraStreaming/src/Screens/Auth/index.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import {GoogleSigninButton} from '@react-native-google-signin/google-signin';
+import React, {FC} from 'react';
```
Comment (evgenyshenets91) [inline agoraStreaming/src/Screens/Auth/index.tsx:2]: please move 2nd line on the first place, cause it is React and will be better to write it firstt

File: agoraStreaming/src/Screens/Auth/helpers/googleSignIn.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import auth from '@react-native-firebase/auth';
+import {GoogleSignin} from '@react-native-google-signin/google-signin';
+import {Dispatch, SetStateAction} from 'react';
```
Comment (evgenyshenets91) [inline agoraStreaming/src/Screens/Auth/helpers/googleSignIn.ts:3]: Are we using here Dispatch and SetStateAction ?

File: agoraStreaming/src/Screens/Home/index.tsx [hunk 6]
```diff
@@ -110,6 +120,11 @@ export const Home: FC<HomeScreenProps> = () => {
           zoomControlEnabled={true}>
           {allMarkers}
         </MapView>
+        <TouchableOpacity onPress={logoutHandler}>
```
Comment (evgenyshenets91) [inline agoraStreaming/src/Screens/Home/index.tsx:123]: please add TODO here that it will be moved on the profile screen

**Review by koverchik — APPROVED** [review]

**Pull request merged**